### PR TITLE
[OpenXR] Blend mode passthrough

### PR DESCRIPTION
**Pull request by svillar**
This PR generalizes the passthrough code by means of an strategy pattern. The different strategies allow us to do:
1. no passthrough
1. passthrough via FB extension
1. passthrough via `XrFrameEndInfo.environmentBlendMode`

Some others could be added in the future too. The first step was to migrate the current code (option 2.) to the strategy pattern. Then we added option 3. It basically adds support for doing passthrough by setting the `XR_ENVIRONMENT_BLEND_MODE_ALPHA_BLEND` to the `environmentBlendMode` attribute of the `XrFrameEndInfo` structure. That works fine for Lynx so this PR is also enabling passthrough for Lynx as well via this new strategy.

### CODE REVIEW

File: app/src/openxr/cpp/DeviceDelegateOpenXR.cpp [hunk 1]
```diff
@@ -1345,22 +1314,29 @@ DeviceDelegateOpenXR::CreateLayerEquirect(const VRLayerPtr &aSource) {
 
 VRLayerPassthroughPtr
 DeviceDelegateOpenXR::CreateLayerPassthrough() {
-  if (!m.layersEnabled || !m.passthroughSupported || !m.passthrough) {
+  if (!m.layersEnabled || !m.passthroughStrategy->usesCompositorLayer()) {
```
Comment (javifernandez) [inline app/src/openxr/cpp/DeviceDelegateOpenXR.cpp:1317]: nit: would make sense to add a check to assert the passthoughStrategy instance is valid ? It seems you do it on other methods as well (eg,  usesPassthroughCompositorLayer)

File: app/src/openxr/cpp/DeviceDelegateOpenXR.cpp [hunk 2]
```diff
@@ -1421,7 +1397,9 @@ DeviceDelegateOpenXR::EnterVR(const crow::BrowserEGLContext& aEGLContext) {
   m.InitializeViews();
   m.InitializeImmersiveDisplay();
   m.InitializeRefreshRates();
-  m.InitializePassthrough();
+
+  m.passthroughStrategy->initializePassthrough(m.session);
```
Comment (javifernandez) [inline app/src/openxr/cpp/DeviceDelegateOpenXR.cpp:1401]: Ditto

File: app/src/openxr/cpp/OpenXRPassthroughStrategy.cpp [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    return !mIsInErrorState && passthroughHandle != XR_NULL_HANDLE;
+}
+
+OpenXRLayerPassthroughPtr
+OpenXRPassthroughStrategyFBExtension::createLayerIfSupported(VRLayerPassthroughPtr vrLayer) const {
+    return OpenXRLayerPassthrough::Create(vrLayer, passthroughHandle);
+}
+
+XrEnvironmentBlendMode
+OpenXRPassthroughStrategyNone::environmentBlendModeForPassthrough() const {
```
Comment (javifernandez) [inline app/src/openxr/cpp/OpenXRPassthroughStrategy.cpp:73]: Just a suggestion; perhaps we could come with an strategy name more descriptive than "none". What about OpenXRPassthroughStrategyNotsupported ?

File: app/src/openxr/cpp/OpenXRPassthroughStrategy.h [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+protected:
+bool mIsInErrorState { false };
+};
+
+class OpenXRPassthroughStrategyNone : public OpenXRPassthroughStrategy {
+private:
+XrEnvironmentBlendMode environmentBlendModeForPassthrough() const override;
+};
+
+class OpenXRPassthroughStrategyFBExtension : public OpenXRPassthroughStrategy {
```
Comment (javifernandez) [inline app/src/openxr/cpp/OpenXRPassthroughStrategy.h:32]: Not a strong opinion, but perhaps we could use independent files for the different strategies. IF we do this, they are more explicit when using source code indexers (or just looking at the source files in our terminal).

**Review by javifernandez — APPROVED** [review]
The change¡looks really nice. Just some minor suggestions that don't affect at all to the logic of the patch and don't spot any defect or error; hence feel free and land the patch as it is if you want.

File: app/src/openxr/cpp/OpenXRPassthroughStrategy.h (hunk 4 above)
Comment (svillar) [reply app/src/openxr/cpp/OpenXRPassthroughStrategy.h:32]: Sure, but the amount of code for some of them is basically 0, that's why I think as a first step it's more convenient to have them all together

File: app/src/openxr/cpp/DeviceDelegateOpenXR.cpp (hunk 2 above)
Comment (svillar) [reply app/src/openxr/cpp/DeviceDelegateOpenXR.cpp:1401]: Same

**Pull request merged**